### PR TITLE
3221 back button

### PR DESCRIPTION
**Pull request by ArashDai**
having an async issue where previousRoute remains undefined. solves #3221

### CODE REVIEW

**Comment by yuviii** [comment]
@ArashDai I think it supposed to take me to the feedback list as i navigated through the feedback, but it is routing to the list for eg if it is course detail then to course list or same case for resource as well.

[Screen navigating video here on this link](https://www.useloom.com/share/f4195306f85a41e1b61fd586adaa576e)

File: src/app/shared/back/back.component.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  selector: 'planet-back',
+  templateUrl: './back.component.html',
+  styleUrls: [ './back.component.scss' ]
+})
+export class BackComponent {
+
+  previousRoute: any;
+
+  constructor(private router: Router) {
+    this.router.events.pipe(
```
Comment (paulbert) [inline src/app/shared/back/back.component.ts]: @ArashDai Is there a reason you are subscribing to router events here?

There is a `Location` service built in to Angular with a `back()` method which should do the same.

**Comment by paulbert** [comment]
@yuviii The back component is only implemented on the `courses/view` route for this PR (we can implement across other routes in another PR).

File: src/app/shared/back/back.component.ts (hunk 1 above)
Comment (ArashDai) [reply src/app/shared/back/back.component.ts]: @paulbert I guess not, I didn't realize the Location service had a back method, I thought I had to go through the router to get to the last page.

File: src/app/shared/back/back.component.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { Component } from '@angular/core';
+import { Location } from '@angular/common';
+
+@Component({
+  selector: 'planet-back',
+  templateUrl: './back.component.html',
```
Comment (lmmrssa) [inline src/app/shared/back/back.component.ts]: It is single line in template so you should have inline templating.

File: src/app/shared/back/back.component.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  styleUrls: [ './back.component.scss' ],
```
Comment (lmmrssa) [inline src/app/shared/back/back.component.ts]: It is empty style file so please remove this file

**Comment by ArashDai** [comment]
I implemented the back button throughout the app, I left old goBack() functions alone if they were used by other pieces of code to not break things. also I left it out of users-achivements because it would take users back into the form if they made an edit and submitted it, then try to go back

**Comment by paulbert** [comment]
I'm going to close this.  Unfortunately it's not a priority to figure out how best to handle the back button navigation in special situations, and we don't have a good solution right now.

**Pull request closed without merging**